container-di/src/main/java/com/yahoo/container/di/Container.java

```diff
@@ -0,0 +1,273 @@
+            }
+        }
+    }
+
+    public void installBundles(Map<ConfigKey<? extends ConfigInstance>, ConfigInstance> configsIncludingBootstrapConfigs) {
+        BundlesConfig bundlesConfig = getConfig(bundlesConfigKey, configsIncludingBootstrapConfigs);
+        osgi.useBundles(bundlesConfig.bundle());
+    }
+
+    private ComponentGraph createComponentsGraph(Map<ConfigKey<? extends ConfigInstance>, ConfigInstance> configsIncludingBootstrapConfigs,
+            long generation, Injector fallbackInjector) {
+
+        previousConfigGeneration = generation;
+
+        ComponentGraph graph = new ComponentGraph(generation);
+
+        ComponentsConfig componentsConfig = getConfig(componentsConfigKey, configsIncludingBootstrapConfigs);
+        if (componentsConfig == null) {
+            throw new ConfigurationRuntimeException("The set of all configs does not include a valid 'components' config. Config set: "
+                    + configsIncludingBootstrapConfigs.keySet());
+        }
+        addNodes(componentsConfig, graph);
+        injectNodes(componentsConfig, graph);
+
+        graph.complete(fallbackInjector);
+        return graph;
+    }
+
+    private void addNodes(ComponentsConfig componentsConfig, ComponentGraph graph) {
+
+        for (ComponentsConfig.Components config : componentsConfig.components()) {
+            BundleInstantiationSpecification specification = bundleInstatiationSpecification(config);
+            Class<?> componentClass = osgi.resolveClass(specification);
+            Node componentNode;
+
+            if (RestApiContext.class.isAssignableFrom(componentClass)) {
+                Class<? extends RestApiContext> nodeClass = componentClass.asSubclass(RestApiContext.class);
+                componentNode = new JerseyNode(specification.id, config.configId(), nodeClass, osgi);
+            } else {
+                componentNode = new ComponentNode(specification.id, config.configId(), componentClass, null);
+            }
+            graph.add(componentNode);
+        }
+    }
+
+    private void constructComponents(ComponentGraph graph) {
+        graph.nodes().forEach(Node::newOrCachedInstance);
+    }
+
+    public void shutdown(ComponentGraph graph, ComponentDeconstructor deconstructor) {
+        shutdownConfigurer();
+        if (graph != null) {
+            deconstructAllComponents(graph, deconstructor);
+        }
+    }
+
+    public void shutdownConfigurer() {
+        configurer.shutdown();
+    }
+
+    // Reload config manually, when subscribing to non-configserver sources
+    public void reloadConfig(long generation) {
+        subscriberFactory.reloadActiveSubscribers(generation);
+    }
+
+    private void deconstructAllComponents(ComponentGraph graph, ComponentDeconstructor deconstructor) {
+        graph.allComponentsAndProviders().forEach(deconstructor::deconstruct);
+    }
+
+    private static final Logger log = Logger.getLogger(Container.class.getName());
```

Comment:
Consider moving the field declaration to top of file.

Reply:
Ahh, missed that one. Fixed.